File: Python/RecursionPractice/reverse_string/reverse_string.py

class RecursivelyReverseString:
    def __init__(self, input_string):
        self.input_string = input_string

    def get_output(self):
        if self.input_string == "":
            return self.input_string
        substring = self.input_string[0]
        self.input_string = self.input_string[1:]
        return self.get_output() + substring

File: Java/RecursionPractice/src/strings/ReverseString.java
package strings;

import java.util.Objects;

public class ReverseString {

    String string;

    public String getOutput() {
        if (Objects.equals(string, "")) {
            return string;
        }
        String substring = String.valueOf(string.charAt(0));
        this.string = string.substring(1);
        return this.getOutput() + substring;
    }

    public ReverseString(String string) {
        this.string = string;
    }
}
